filament/src/PostProcessManager.cpp

```diff
@@ -1919,45 +1919,64 @@ PostProcessManager::BloomPassOutput PostProcessManager::bloomPass(FrameGraph& fg
             [=](FrameGraphResources const& resources,
                     auto const& data, DriverApi& driver) {
 
-                auto const& material = getPostProcessMaterial("bloomDownsample");
-                FMaterialInstance* mi = material.getMaterialInstance(mEngine);
-
-                const PipelineState pipeline(material.getPipelineState(mEngine));
-
                 auto hwIn = resources.getTexture(data.in);
                 auto hwOut = resources.getTexture(data.out);
                 auto hwStage = resources.getTexture(data.stage);
 
-                mi->use(driver);
-                mi->setParameter("source", hwIn, {
+                auto const& material = getPostProcessMaterial("bloomDownsample");
+                auto const* ma = material.getMaterial(mEngine);
+
+                FMaterialInstance* mis[] = {
+                        ma->createInstance("bloomUpsample-ping"),
+                        ma->createInstance("bloomUpsample-pong"),
+                        ma->createInstance("bloomUpsample-first"),
```

Comment:
Should these be named `bloomDownsample-*`?

Reply:
ah good catch